shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -154,10 +156,34 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 
 - (void)showShareViewController:(NSString*)content {
   UIViewController* engineViewController = [_engine.get() viewController];
+
   NSArray* itemsToShare = @[ content ?: [NSNull null] ];
   UIActivityViewController* activityViewController =
       [[[UIActivityViewController alloc] initWithActivityItems:itemsToShare
                                          applicationActivities:nil] autorelease];
+
+  if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
+    // TODO(louisehsu): consider moving this to TextInputPlugin as functionality is very similar
+    // On iPad, the share screen is presented in a popover view, and requires a CGRect
+    FlutterTextInputPlugin* _textInputPlugin = [_engine.get() textInputPlugin];
+    UITextRange* range = _textInputPlugin.textInputView.selectedTextRange;
+
+    CGRect firstRect = [(FlutterTextInputView*)_textInputPlugin.textInputView
+        caretRectForPosition:(FlutterTextPosition*)range.start];
+    CGRect transformedRectLeft = [(FlutterTextInputView*)_textInputPlugin.textInputView
+        localRectFromFrameworkTransform:firstRect];
+    CGRect lastRect = [(FlutterTextInputView*)_textInputPlugin.textInputView
+        caretRectForPosition:(FlutterTextPosition*)range.end];
+    CGRect transformedRectRight = [(FlutterTextInputView*)_textInputPlugin.textInputView
+        localRectFromFrameworkTransform:lastRect];
+
+    activityViewController.popoverPresentationController.sourceView = engineViewController.view;
+    activityViewController.popoverPresentationController.sourceRect =
+        CGRectMake(transformedRectLeft.origin.x, transformedRectLeft.origin.y,
+                   transformedRectRight.origin.x - transformedRectLeft.origin.x,
+                   transformedRectLeft.size.height);
+  }
```

Comment:
this won't work for RTL language. possibly crash due to negative width

Reply:
hmm, would it help if i just wrapped in abs() ? I dont know what rtl language is

Reply:
RTL = right-to-left, e.g. arabic/hebrew